pandas/core/arrays/categorical.py

```diff
@@ -628,8 +630,23 @@ def from_codes(cls, codes, categories, ordered=False):
             categorical. If not given, the resulting categorical will be
             unordered.
         """
+        codes = np.asarray(codes)
+        if not is_integer_dtype(codes):
+            err = True
+            if is_float_dtype(codes):
+                icodes = codes.astype('i8')
+                if (icodes == codes).all():
+                    err = False
+                    codes = icodes
+                    warn("float codes will be disallowed in the future",
+                         FutureWarning)
+                else:
+                    err = True
+            if err:
```

Comment:
instead of using ``err``, you can create the message say around line 635 like
``msg = ......``

and then just ``raise ValueError(msg)`` where needed, much easier to read

Reply:
```python
        if not is_integer_dtype(codes):
            msg = "codes need to be array-like integers"
            if is_float_dtype(codes):
                icodes = codes.astype('i8')
                if (icodes == codes).all():
                    err = False
                    codes = icodes
                    warn("float codes will be disallowed in the future",
                         FutureWarning)
                else:
                    raise ValueError(msg)
            else:
                raise ValueError(msg)
```

is that what you want? That doesn't read easier to me, but I can change it.

Reply:
I prefer the `msg` way to the `err` way as well.